vllm/model_executor/models/llama.py

```diff
@@ -270,6 +270,30 @@ def forward(
 
 class LlamaForCausalLM(nn.Module):
     supports_lora = True
+    lora_target_modules = [
```

Comment:
The biggest question I have about this PR is these configurations, I don't think they should be here. There might be different ways to run a model (e.g. different layers get LoRAified, like for the mixtral model, do you apply LoRA to the MoE weight matrices or not?), and also it seems odd to have the lora configurations in the model itself. Are there better places where this can fit? Maybe @Yard1 has some ideas here as well :)

The other issue of the override code in `vllm/lora/models.py` will flow naturally from what we do here.

Reply:
Here are my thoughts so far: The `lora_target_modules` I believe is in the `adapter_config.json` of the LoRA model file, so we probably can use the information from there -- it just will need to be adapted a little bit via the `packed_module_mapping`. So it seems we can remove the LoRA specific stuff, which is already a big relief.

The other parameters are also useful for loading the model etc, so maybe we should keep them here? It would be worth investigating if we can reuse this in the model loading function for the base model (`load_weights`).

Reply:
This defines all the layers we support for LoRA on the given model. The adapter can use a subset of those layers. We should not be using `adapter_config.json` directly because:
1. different adapters will have different layers, therefore we need a common superset
2. that superset has to be constant
3. we need to implement support for each layer type, so if an adapter specifies a layer we do not support, we need to throw exceptions.

Reply:
Sounds great, I believe my confusion will go away if we rename `lora_target_modules` to `supported_lora_modules` so it is clearer that this is the superset of modules that we support, and not the actual ones that are loaded. Let's rename it, and in parallel I'll make a PR that shows how it would look like if we use `packed_module_mapping` in the `load_weights` function so the information is in only one place.

Let's also reorder things so the generic attributes come before the lora specific ones (and maybe add a comment on the lora specific ones that they are lora specific).

Reply:
I put up a PR in https://github.com/vllm-project/vllm/pull/2843 :)